clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -181,187 +169,35 @@
  * Any consumer that uses sticky assignment can leverage this listener like this:
  * <code>consumer.subscribe(topics, new TheNewRebalanceListener());</code>
  *
+ * Note that you can leverage the {@link CooperativeStickyAssignor} to never revoke any partitions to begin with, except
```

Comment:
The StickyAssignor javadocs include a cooperative rebalancing hack using the rebalance listener -- should we remove it or just point to the CooperativeStickyAssignor as the recommended way to achieve this?

Reply:
I think we would remove the `StickyAssignor` in the future if we realized that there's rare use cases that needs a eager sticky assignor; until then I'm fine with the current way as is.